README
TAILS Candy Ver. 0.2
********************

	(C) 2015-2017 by Dirk Praet

License: GPL v3 - http://www.gnu.org/licenses/gpl.txt

I - DESCRIPTION
=   ===========

"TAILS Candy" is a small software package designed for use with TAILS, The
Amnesic Incognito Live System. It is not meant to be used with any other OS and
REQUIRES a TAILS-installation with a persistent volume and Dotfiles support
configured. 
See https://tails.boum.org/doc/first_steps/persistence/configure/index.en.html .

The package consists of a number of scripts that add additional features to 
TAILS, i.e. on-demand installations/upgrades of some useful software that does
not come with a default TAILS stock installation. Most of these scripts will
require the root password to be configured at the TAILS Greeter login screen.

The following are currently included:

1) Academic Signature v55 - A Swiss army knife of (ECC) encryption
   http://www.fh-wedel.de/~an/crypto/Academic_signature_eng.html

2) AES Crypt 3.10 - Advanced File Encryption
   http://www.aescrypt.com

3) AIDE 0.16 - Advanced Intrusion Detection Environment
   http://aide.sourceforge.net

4) Armory 0.93.3 - A Bitcoin wallet manager
   https://bitcoinarmory.com

5) Dropbox 4.49 - Cloud storage
   https://www.dropbox.com

6) GostCrypt 1.3 - Truecrypt successor with GOST-support
   http://www.gostcrypt.org/

    *** Broken on current TAILS 2.5 because of WxWidgets 3.0
    *** conflicts. (same as Academic Signature)

7) Horcrux - Duplicity based, GPG encrypted backup/restore script
   https://github.com/piffio/horcrux
   http://chrispoole.com/project/general/horcrux/

8) Jitsi 2.8 - SIP Communicator
   https://jitsi.org

9) Linux Malware Detect 1.5 - Malware detection & monitoring
   https://www.rfxn.com/projects/linux-malware-detect/

10) Mumble 1.2.8 - Low latency VOIP client
    http://www.mumble.com

11) Martus 4.4.0/5.1.1 - Secure information collection and management system
    https://www.martus.org

12) OnionMail 1.9.0 - Set up (a) secure email account(s) on an OnionMail server.
    http://en.onionmail.info

13) OnionShare 0.9.1 - Secure file sharing
    https://onionshare.org/

    *** Broken on current TAILS 2.5 . Awaiting solution from Tor/TAILS
    *** developers                                                    

14) OpenSSH - Upgrade moduli file to 4096-bit values
    http://www.openssh.org

15) Paranoia Text & File Encryption 12R1C
    http://www.paranoiaworks.mobi

16) Pond 0.1.1 - Forward secure, asynchronous messaging 
    https://pond.imperialviolet.org

    *** Broken on current TAILS 2.5 . ***

17) PyBitmessage 0.4.4 - P2P encrypted messaging (with opt. Mixmaster support)
    https://github.com/Bitmessage/PyBitmessage

18) Rootkit Hunter 1.4.2 - Linux rootkit detection
    https://rootkit.nl/projects/rootkit_hunter.html

19) Tahoe LAFS 1.10 - Encrypted distributed file system (with or without I2P)
    https://www.tahoe-lafs.org/

20) TeamViewer 11 - Remote support
    https://www.teamviewer.com/en/

21) Tomb 2.2 - LUKS container creation and management
    https://www.dyne.org/software/tomb/

22) TorChat 0.9.9-553 - Secure instant messaging over Tor
    https://github.com/prof7bit/TorChat

23) Tor Messenger 0.2.0b2 - Secure instant messaging over Tor (InstantBird)
    https://blog.torproject.org/blog/tor-messenger-beta-chat-over-tor-easily

24) Shamir's Secret Sharing Scheme - Python GUI for SSSS
    https://github.com/skhakimov/secret-sharing

25) Silent Eye 0.4.0 - Steganography tool
    http://www.silenteye.org/

26) TrueCrypt 7.2 - Truecrypt
    http://sourceforge.net/projects/truecrypt/

27) Veracrypt 1.18 - Truecrypt/Veracrypt encrypted container management
    https://veracrypt.codeplex.com/

28) VLC 2.2.4 - VLC Media Player
    https://www.videolan.org

    *** Broken on current TAILS 2.5   . ***
    *** due to conflicting dependencies ***

29) Zulucrypt 5.0.1 - Cryptsetup, TCPlay, LUKS & EncFS management
    https://mhogomchungu.github.io/zuluCrypt/

30) Yubikey Personalization Manager 3.1.16 - Authentication token management
    https://yubico.com

31) OnionCat 0.2.2 - IP-Transparent Tor hidden service connector
    https://www.onioncat.org
    

None of these packages come with "TAILS Candy". They are fetched on demand
from Debian repositories, cloned/pulled from Github or downloaded from their
respective sites.

Many of the installation scripts are based on TAILS-installation instructions
or wrappers around routines already provided by the respective authors of the
packages (in order not to break stuff when they are revised or upgraded).

In addition to these scripts, "TAILS Candy" provides a couple of additional
enhancements in the form of login hooks, Gnome desktop and other modifications
to allow for further customisation. 


II - CONTENTS
==   ========

"TAILS Candy" will install two folders in the TAILS ~/Persistent folder:

- Packages  : Contains the actual installation/upgrade scripts as well as 
              subfolder called "Repo" and another one called "Settings".
              Repo will store downloaded packages for subsequent installations.
              Settings stores package configuration settings that cannot be
              moved to the Dotfiles persistent directory.
- TAILSCandy: A folder with .desktop files to execute the installation/upgrade
              scripts in Packages with a simple point-and-click.

The hooks and modifications previously mentioned will all be taken care of
by the "TAILS Candy" installer. Here's a short summary:

1) Create a persistent autostart entry in ~/.config/autostart that will be
executed every time you log in to TAILS. You can customise the Customisations.sh
script it runs and which can be found in the ~/Persistent/Packages folder.

2) Disable TAILS Gnome Keyring Daemon SSH/GPG-support by overriding default
autostart. This is done by creating another ~/.config/autostart item which is
a copy of the one in /etc/xdg/autostart but with Autostart set to false.
Reason for doing so is that Gnome Keyring s*cks, doesn't support recent SSH key
types or smart cards, and normal people are using the GnuPG agent for this
purpose anyway.

3) Check for GPG SSH-support in ~/.bashrc and ~/.gnupg/gpg-agent.conf. Set if
not found. Check if you have configured GPG persistence.

4) Persistently modify the Gnome menu to contain an "Encryption" group menu
item ( ~/.config/menus/ and ~/.local/share/desktop-directories/Encryption.
directory )

5) Disable weak DHE-exchanges & RC4-ciphers in the user's Tor Browser default
profile.

6) Create a link to the TAILSCandy folder on the user's desktop so he/she can
perform point-and-click installations/updates.

7) Propose to make I2P settings persistent.

8) Ask if you want to make some additional Debian package installations
persistent by adding them to the live-additional-software.conf file (e.g.
Mixmaster, FileZilla, Putty).


III - REQUIREMENTS
===   ============

1) A working TAILS 2.5 with persistent volume and Dotfiles support.
2) Root password set up at the TAILS Greeter (login) screen.
3) Some patience and a kind disposition. This is an early release.
   Most (bash) scripts are really basic-basic and will benefit from some
   optimisation and code clean-up. This will happen over time.
   They will also most probably contain some left-over bugs here and there.


IV - (DE)INSTALLATION and UPGRADE
==   ============================

Make the install_candy.sh script executable (chmod u+x) and run from where ever 
you have uncompressed the .tar.gz distribution file. The installer will move the
Packages and TAILSCandy folders to your ~/Persistent directory, guide you
through the installation and clean up the installation directory package.
Reinstallations or upgrades can be done through the TAILS menu (Applications,
Other).
 
If you're tired of TAILS Candy, the package can be uninstalled by making the
uninstall script remove_candy.sh executable and then running it as root. It
will remove the two main folders from ~/Persistent and undo (most of) the
modifications described earlier. The uninstaller will ask you to either keep or
remove any applications previously installed through TAILS Candy. For those you
remove you will be asked whether or not to preserve their application settings
stored in ~/Persistent/Packages/Settings. If answered yes to, they will be saved
in ~/Persistent/TAILS_Candy_Saved_Settings . You're on your own to reconfigure
the preserved applications with these. Simplest way would probably be to move
that folder back to ~/Persistent/Packages/Settings .


V - CAVEATS
=   ======= 

1) None of the packages that can be installed or upgraded through TAILS Candy
should compromise TAILS security, and those requiring network access do so
over Tor (some of which need manual socks5 proxy configuration to tunnel
over localhost:9050). That said, I can in no way guarantee that this is not the
case, so any risks relating to installation and use of any package you install
through TAILS Candy are yours and yours alone.

2) At this time, TAILS Candy does NOT come with AppArmor profiles for the
additional packages it can install. Perhaps in some future release.

3) Do NOT add additional stuff just for the heck of it. Everything you
install increases the system's attack surface, so install only what you need in any given TAILS session. This goes for both the live-additional-software.conf
file (for persistent TAILS package installs) and the TAILS Candy packages. The
former also (considerably) adds to boot time.

4) The TAILS Candy scripts only guide you through installation/upgrade of the
supported packages. They don't explain you how to set them up or use them.
So if you're not familiar with them, RTFM or visit the reference URL's given
above.

5) All the usual restrictions on TAILS and Tor apply. If used correctly,
they should provide you with some increased degree of privacy, security and
anonimity - especially when compared to a stock Windows installation - but they
will not protect you from resourceful state actors or hardware-based attacks (
evil maid, compromised chips, BIOS/UEFI infestations etc.).

You can still mitigate the latter by running TAILS on older hardware stuffed
with RAM and preferably without wireless, bluetooth and sound chips. Unless you
have good reasons to use the TAILS persistence feature, running from DVD really 
is more secure too. If you know what you're doing, you can also run from DVD
and access a persistent TAILS LUKS-volume on an USB-stick or SSD card.


VI - KNOWN ISSUES
==   ============

1) If you have installed Dropbox, for mysterious reasons it loses its linking
information on every reboot and will ask you to register again. 
2) Minor issues with Tomb panel applet and system mime type registration
3) OnionMail was temporarily removed from TAILS Candy because it still requires
ClawsMail while IceDove is now the default on TAILS.


VII - COMMENTS
===   ========
All comments and suggestions are welcome. You can also drop me a note at
skylord|at|jedi|dot|be if you want your application included in this package.
Security-Anonimity-Privacy stuff only, please.



Happy TAILS'ing, stay safe and s*d the FiveEyes !


NEWS

- Commit of a number of modifications for TAILS 2.5
- Current TAILS 2.5 has a number of issues that prevent Academic Signature 
from compiling or VLC from installing due to conflicting dependencies.


remove_candy.sh
#!/bin/bash
#########################################################################
# TAILS Candy removal script
#
# This script removes the TAILS Candy package
#
# Part of "TAILS Candy" Project
# Version 0.1a
# License: GPL v3 - Copy included in distribution package
#
# By Dirk Praet - skylord@jedi.be
#########################################################################

# Function declarations
# #####################

function error_exit
{
        echo "$1" 1>&2
        exit 1
}

Confirm() { read -sn 1 -p "$* [Y/N]? "; [[ ${REPLY:0:1} = [Yy] ]]; }

SaveConfig()
{
savedsettings=$PERSISTENT/TAILS_Candy_Saved_Settings
mv $PERSISTENT/Packages/Settings $savedsettings
cd $savedsettings
mv $savedsettings/Gnome/panel-default-layout.layout.saved .. 1>&2
rm -rf Gnome 1>&2
echo "Settings saved in $savedsettings"
echo
cd -
}

RemovePackage()
{
grep -vFf <(echo "$1") $softfile > /tmp/brol
mv /tmp/brol $softfile
chmod 600 $softfile
chown tails-persistence-setup:tails-persistence-setup $softfile
}

RemoveI2P()
{
grep -vFf <(echo "/var/lib/i2p source=i2p") /live/persistence/TailsData_unlocked/persistence.conf > /tmp/brol
chmod 600 /live/persistence/TailsData_unlocked/persistence.conf
chown tails-persistence-setup:tails-persistence-setup /live/persistence/TailsData_unlocked/persistence.conf
}

####################
# Script main line #
####################

if [[ $EUID -ne 0 ]]; then
    echo "You need to run this script as root, i.e. sudo ./remove_candy.sh ." 1>&2
    exit 1
fi

clear
echo 
echo "This script will remove the TAILS Candy package."
echo 
echo "Packages you have installed with TAILS Candy will be preserved and"
echo "you can choose to keep their configuration settings."
echo
read -n 1 -p "Press any key to continue or Ctrl-C to abort ..."

DOT_DIR=/live/persistence/TailsData_unlocked/dotfiles
PERSISTENT=/home/amnesia/Persistent

echo
echo "Removing Packages and TAILSCandy folders in $PERSISTENT ..."
echo
Confirm "Do you wish to save application configuration settings? y/n" && SaveConfig 
rm -rf $PERSISTENT/Packages $PERSISTENT/TAILSCandy

echo "Removing start-up hook in ~/.config/autostart/"
rm -f $DOT_DIR/.config/autostart/customisations.desktop 1>&2
rm -f /home/amnesia/.config/autostart/customisations.desktop 1>&2
echo "Removing deactivation of SSH Gnome Keyring support in ~/.config/autostart/"
rm -f $DOT_DIR/.config/autostart/gnome-keyring-ssh.desktop 1>&2
rm -f /home/amnesia/.config/autostart/gnome-keyring-ssh.desktop 1>&2

echo "Leaving GnuPG SSH support in ~/.bashrc ..."
echo "Leaving SSH support in ~/.gnupg/gpg-agent.conf ..."
echo "Leaving Gnome menu customisations in ~/.config/menus to preserve Encryption submenu ..."

echo "Removing TAILS Candy folder from desktop ..."
rm -f /home/amnesia/Desktop/TAILS_Candy.desktop

echo
grep -q "Packages" /live/persistence/TailsData_unlocked/dotfiles/.config/gnome-panel/panel-default-layout.layout
if [ $? -eq 0 ]; then
echo "Removing custom TAILS Candy Gnome top panel..."
rm -f /live/persistence/TailsData_unlocked/dotfiles/.config/gnome-panel/panel-default-layout.layout
fi 

#
# Remove I2P persistence?
#
grep -q "I2P" /live/persistence/TailsData_unlocked/persistence.conf
if [ $? -eq 0 ]; then
echo
Confirm "Would you like to remove I2P persistence? " && RemoveI2P
fi

# Removing applications installed by TAILS Candy
cd $PERSISTENT

# Academic Signature
cnt=`ls -d aca_sig*/ 2>/dev/null | wc -l`
if [ "$cnt" = "1" ]; then
appdir=`ls -d aca_sig*/`
echo
Confirm "Would you like to remove the Academic Signature installation? " && rm -rf $PERSISTENT/$appdir
echo
fi

# Dropbox
appdir=$DOT_DIR/.dropbox-dist
if [ -d $appdir ]; then
echo
Confirm "Would you like to remove the Dropbox installation? " && rm -rf $DOT_DIR/.dropbox*
fi

# Onionshare
appdir=$PERSISTENT/onionshare
if [ -d $appdir ]; then
appdir="/home/amnesia/Persistent/onionshare /home/amnesia/Persistent/.onionshare_install"
echo
Confirm "Would you like to remove the Onionshare installation? " && rm -rf $appdir
fi

# PyBitmessage
appdir=$PERSISTENT/PyBitmessage
if [ -d $appdir ]; then
echo
Confirm "Would you like to remove the PyBitmessage installation? " && rm -rf $appdir
fi

# Tor Messenger
appdir=$PERSISTENT/tor-messenger
if [ -d $appdir ]; then
echo
Confirm "Would you like to remove the Tor Messanger (InstantBird) installation? " && rm -rf $appdir
fi

# Tomb
appdir=$PERSISTENT/Tomb
if [ -d $appdir ]; then
echo
Confirm "Would you like to remove the Tomb installation? " && rm -rf $appdir
fi

# Linux Malware Detect
cnt=`ls -d maldetect*/ 2>/dev/null | wc -l`
if [ "$cnt" = "1" ]; then
appdir=`ls -d maldetect*/`
echo
Confirm "Would you like to remove the Linux Malware Detect installation? " && rm -rf $PERSISTENT/$appdir
echo
fi

# Martus Client
cnt=`ls -d Martus*/ 2>/dev/null | wc -l`
if [ "$cnt" = "1" ]; then
appdir=`ls -d Martus*/`
echo
Confirm "Would you like to remove the Martus Client installation? " && rm -rf $PERSISTENT/$appdir
echo
fi

# SSSS GUI
appdir=$PERSISTENT/SSSS
if [ -d $appdir ]; then
echo
Confirm "Would you like to remove the SSSS GUI installation? " && rm -rf $appdir
fi

#
# Remove additional software packages from /live/persistence/TailsData_unlocked/live-additional-software.conf
#

softfile=/live/persistence/TailsData_unlocked/live-additional-software.conf
packages='dirmngr gpgsm claws-mail-smime-plugin pidgin-openpgp signing-party clamtk claws-mail-clamd-plugin filezilla putty mixmaster encfs cryptkeeper pidgin-privacy-please proxychains nmap hfsprogs'
for package in $packages
do
	grep -q $package $softfile
	if [ $? -eq 0 ]; then
		echo
		Confirm "Do you wish to remove $package from live-additional-software.conf? " && RemovePackage $package
	fi
done
echo

sudo -u amnesia /usr/bin/notify-send "TAILS Candy succesfully removed" "Reboot for changes to take effect."



install_candy.sh
#!/bin/bash
#########################################################################
# TAILS Candy installer script
#
# This script installs the TAILS Candy package
#
# Part of "TAILS Candy" Project
# Version 0.2
# License: GPL v3 - Copy included in distribution package
#
# By Dirk Praet - skylord@jedi.be
#########################################################################

# Function declarations
# #####################

function error_exit
{
        echo "$1" 1>&2
        exit 1
}

function modify_toppanel
{
	echo "Replacing default TAILS Gnome top panel..."
	l1dir="/live/persistence/TailsData_unlocked/dotfiles/.config/gnome-panel"
	l2dir="/home/amnesia/.config/gnome-panel"
	if [ ! -d "$l1dir" ]; then
		sudo -u amnesia mkdir -p $l1dir
	fi
	if [ ! -d "$l2dir" ]; then
		sudo -u amnesia mkdir -p $l2dir
	fi
	layoutfile="/live/persistence/TailsData_unlocked/dotfiles/.config/gnome-panel/panel-default-layout.layout"
	layoutfile2="/home/amnesia/.config/gnome-panel/panel-default-layout.layout"
	layoutsave=$PERSISTENT/Packages/Settings/Gnome/panel-default-layout.layout.saved
	if [ -f "$layoutfile" ]; then
		sudo -u amnesia cp -p $layoutfile $layoutsave
	else
	if [ -f "$layoutfile2" ]; then
		sudo -u amnesia cp -p $layoutfile2 $layoutsave
	fi
	fi
	sudo -u amnesia cp -p $PERSISTENT/Packages/Settings/Gnome/panel-default-layout.layout $layoutfile
	if [ -f "$layoutsave" ]; then
		echo "Original panel layout saved as $layoutsave"
	fi
}

Confirm() { read -sn 1 -p "$* [Y/N]? "; [[ ${REPLY:0:1} = [Yy] ]]; }

####################
# Script main line #
####################

if [[ $EUID -ne 0 ]]; then
    echo "You need to run this installer script as root" 1>&2
    exit 1
fi

CUR_DIR=$PWD
DOT_DIR=/live/persistence/TailsData_unlocked/dotfiles
cd $DOT_DIR || error_exit "No TAILS Dotfiles persistence found. Aborting"
cd "$CUR_DIR"

clear
echo 
echo "This routine will guide you through installation or upgrade"
echo "of TAILS Candy."
echo
read -n 1 -p "Press any key to continue or Ctrl-C to abort ..."

# Determine if this is a first time install or upgrade.
#
PERSISTENT=/home/amnesia/Persistent

packdir=$PERSISTENT/Packages
candydir=$PERSISTENT/TAILSCandy 
if [ -d "$packdir" ] && [ -d "$candydir" ] ; then
# This is an upgrade
	#if [ "$CUR_DIR" == "/home/amnesia/Persistent/Packages" ]; then
		echo "Downloading latest TAILS Candy from Github ..."
		# We are being executed from the Packages directory, so let's
		# update by performing a git clone from the master branch
		sudo -u amnesia mkdir -p $PERSISTENT/tmp  2> /dev/null
		cd $PERSISTENT/tmp
		sudo -u amnesia git clone https://github.com/dpramone/TAILS-Candy.git || error_exit "Unable to download TAILS Candy upgrade. Bailing out ..."
		cd TAILS-Candy
		# Remove Git infornation
		rm -rf /home/amnesia/Persistent/tmp/TAILS-Candy/.git
		CUR_DIR="/home/amnesia/Persistent/tmp/TAILS-Candy"
	#fi
	VERSION=`cat "$CUR_DIR/VERSION"`
	echo
	echo "Upgrading TAILS Candy to version $VERSION ..."
	# 
	# Placeholder for stuff to be removed from older versions ...
	#
else
	VERSION=`cat "$CUR_DIR/VERSION"`
	echo
	echo "Installing TAILS Candy version $VERSION ..."
fi

echo
echo "Installing Packages and TAILSCandy folders in $PERSISTENT ..."
folder="$CUR_DIR/Packages"
if [ -d "$folder" ]
then
	chown -R amnesia:amnesia "$folder"
	cp -rp "$folder" $PERSISTENT/
	if [ ! -f "$PERSISTENT/Packages/install_candy.sh" ]; then
	cp -p "$CUR_DIR/install_candy.sh" $PERSISTENT/Packages/ 2> /dev/null
	fi 
	chmod 700 "$folder"/*.sh
else
	error_exit "Where is our TAILS Candy Packages folder? Installation aborted."
fi
folder="$CUR_DIR/TAILSCandy"
if [ -d "$folder" ]
then
	chown -R amnesia:amnesia "$folder"
	cp -rp "$folder" $PERSISTENT/
	mkdir -p $PERSISTENT/TAILSCandy/Info 2> /dev/null
	cp -p "$CUR_DIR/TODO" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/AUTHORS" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/README" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/LICENSE" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/COPYING" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/ChangeLog" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/NEWS" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	cp -p "$CUR_DIR/VERSION" $PERSISTENT/TAILSCandy/Info/ 2> /dev/null
	chown -R amnesia:amnesia $PERSISTENT/TAILSCandy/Info
else
	error_exit "Where is our TAILSCandy folder? Installation aborted."
fi

# Add persistent start-up hook in ~/.config/autostart/
#
startdir="/live/persistence/TailsData_unlocked/dotfiles/.config/autostart/"
if [ ! -d "$startdir" ]; then
sudo -u amnesia mkdir -p $startdir
fi
sudo -u amnesia cat <<EOF > /live/persistence/TailsData_unlocked/dotfiles/.config/autostart/customisations.desktop
[Desktop Entry]
Encoding=UTF-8
Name=Customisations
GenericName=Customisations
Comment=Customisations hook
Exec=/home/amnesia/Persistent/Packages/Customisations.sh
Terminal=false
Type=Application
X-GNOME-Autostart-enabled=true
EOF
chown amnesia:amnesia $startdir/customisations.desktop
chmod 600 $startdir/customisations.desktop
if [ ! -d "/home/amnesia/.config/autostart" ]; then
	sudo -u amnesia mkdir -p /home/amnesia/.config/autostart
fi
sudo -u amnesia ln -sf $startdir/customisations.desktop /home/amnesia/.config/autostart/customisations.desktop

#
# Disable default TAILS SSH support in Gnome Keyring daemon.
# It totally sucks, doesn't support ECC keys and normal people should
# be using the GnuPG agent anyway.
#
echo "Disabling Gnome Keyring SSH support so we can use the GPG agent ..."
sudo -u amnesia cat <<EOF > /live/persistence/TailsData_unlocked/dotfiles/.config/autostart/gnome-keyring-ssh.desktop
[Desktop Entry]
Encoding=UTF-8
Type=Application
Name=SSH Key Agent
Name[en_GB]=SSH Key Agent
Comment=GNOME Keyring: SSH Agent
Comment[en_GB]=GNOME Keyring: SSH Agent
Exec=/usr/bin/gnome-keyring-daemon --start --components=ssh
OnlyShowIn=GNOME;Unity;
X-GNOME-Autostart-Phase=Initialization
X-GNOME-AutoRestart=false
X-GNOME-Autostart-Notify=true
X-GNOME-Bugzilla-Bugzilla=GNOME
X-GNOME-Bugzilla-Product=gnome-keyring
X-GNOME-Bugzilla-Component=general
X-GNOME-Bugzilla-Version=3.4.1
X-GNOME-Autostart-enabled=false
EOF

chown amnesia:amnesia $startdir/gnome-keyring-ssh.desktop
chmod 600 $startdir/gnome-keyring-ssh.desktop
sudo -u amnesia ln -sf $startdir/gnome-keyring-ssh.desktop /home/amnesia/.config/autostart/gnome-keyring-ssh.desktop 1>&2

echo "Checking .bashrc for GnuPG SSH support ..."
grep -q "GPG_TTY" /home/amnesia/.bashrc
if [ ! $? -eq 0 ]; then
	sudo -u amnesia cat <<EOF >> /home/amnesia/.bashrc
# 
# GnuPG SSH support
# 
GPG_TTY=$(tty)
export GPG_TTY

if [ -f "${HOME}/.gnupg/gpg-agent-info-amnesia" ]; then
     . "${HOME}/.gnupg/gpg-agent-info-amnesia"
       export GPG_AGENT_INFO
       export SSH_AUTH_SOCK
       export SSH_AGENT_PID
fi
EOF
bashrc=/live/persistence/TailsData_unlocked/dotfiles/.bashrc
# Make .bashrc persistent by copying over to Dotfiles directory
if [ ! -f "$bashrc" ]; then
	sudo -u amnesia cp -p /home/amnesia/.bashrc $bashrc
fi
fi

gpgagent="/home/amnesia/.gnupg/gpg-agent.conf"
grep -q "enable-ssh-support" $gpgagent
if [ ! $? -eq 0 ]; then
	sudo -u amnesia echo "enable-ssh-support" >> $gpgagent
fi

#
# Modify Gnome menu/directory entries with TAILS Candy entries
#
echo "Creating a Gnome menu Encryption application group item"
sudo -u amnesia cp -rp $PERSISTENT/Packages/Settings/Gnome/menus /live/persistence/TailsData_unlocked/dotfiles/.config/ 1>&2
dtdir=/live/persistence/TailsData_unlocked/dotfiles/.local/share/desktop-directories
if [ ! -d "$dtdir" ]; then
	sudo -u amnesia mkdir -p $dtdir
fi
sudo -u amnesia cp -p $PERSISTENT/Packages/Settings/Gnome/Encryption.directory $dtdir/Encryption.directory 1>&2

#
# Shall we modify the default TAILS Gnome top panel layout?
#
echo
Confirm "Shall we add a couple of useful items to the Gnome top panel? If you answered yes on previous TAILS 1.6 ot 1.7 instalations, please repeat now to reflect changes in TAILS 1.8 and up." && modify_toppanel

#
# Configure persistence for I2P ?
#
echo
grep -q "i2p" /live/persistence/TailsData_unlocked/persistence.conf
if [ ! $? -eq 0 ]; then
Confirm "Shall we configure TAILS for I2P persistence? " && echo "/var/lib/i2p source=i2p" >> /live/persistence/TailsData_unlocked/persistence.conf
fi
 
#
# Ask for additional software packages to be installed persistently
#
clear
echo
echo "You can now pick a series of other standard packages to be installed"
echo "automatically every time you boot TAILS."
echo 
echo "1) pidgin-privacy-please and pidgin-openpgp are privacy add-ons for"
echo "the Pidgin IM client."
echo "2) ClamTk"
#echo "3) ClamTk and claws-mail-clamd-plugin: forwarding emails containing"
#echo "viruses is never a good idea."
echo "3) Putty and Filezilla are for those folks who cannot be asked to use"
echo "the command line for secure remote sessions and file transfers."
echo "4) encfs and cryptkeeper let you access Dropbox storage encrypted with"
echo "encfs or BoxCryptor on other platforms."
echo "5) Mixmaster and signing-party for die-hard cypherpunks."
echo "6) Proxychains for use with nmap during network reconnaissance."
echo "7) hfsprogs: set of utilities for accessing your Mac's hard drive(s)."
echo

packages='dirmngr gpgsm pidgin-openpgp signing-party clamtk filezilla putty mixmaster encfs cryptkeeper pidgin-privacy-please proxychains nmap hfsprogs'
for package in $packages
do
	grep -q $package /live/persistence/TailsData_unlocked/live-additional-software.conf
	if [ ! $? -eq 0 ]; then
		echo
		Confirm "Do you wish to consistently install $package at boot time? " && echo $package >> /live/persistence/TailsData_unlocked/live-additional-software.conf
	fi
done
echo

#
# Set RAMONES desktop background
#
# Unless you comment out the next 2 statements, your desktop background will get
# a Ramones wallpaper. People REALLY should take their time to go through
# installation routines like this one to see what exactly it is they're doing.
# You're on a SECURE OS, so try to keep it that way.
# The wallpaper is temporary and TAILS will automatically reset it at reboot
#
sudo -u amnesia /usr/bin/gsettings set org.gnome.desktop.background picture-uri file:///home/amnesia/Persistent/Packages/Settings/Gnome/bg.jpg
sudo -u amnesia /usr/bin/gsettings set org.gnome.desktop.background picture-options "scaled"

# Execute Customisations script from TAILS Candy Packages folder. This will from now be executed at every boot
sudo -u amnesia /home/amnesia/Persistent/Packages/Customisations.sh install

read -n 1 -p "All finished. Some modifications will only take effect after reboot. Press any key to finish up ..."

# Remove installation directory
cd "$CUR_DIR"
cd ..
rm -rf ./TAILS-Candy-master
rm -rf ./TAILS-Candy

exit 0

